Docs: Update Stepper docs

cherylpaulsen opened the pull request:
Merging best practice guidance. Updates include:
- Add description
- Reorder content
- Update 'Linear mode' section copy
- Update 'Linear mode' example to include secondary copy
- Update 'Non-linear mode' section copy
- Update 'Non-linear mode' example to include secondary copy and navigation buttons
- Add new section 'Providing user controls' with example
- Add 'When to use' section
- Fix spelling 'Forth' to 'Fourth' in multiple locations

felixhabib commented on packages/braid-design-system/src/lib/components/Stepper/Stepper.docs.tsx:
```diff
@@ -71,20 +83,73 @@ const docs: ComponentDocs = {
   alternatives: [],
   additional: [
+    {
+      label: 'Alignment',
```
While we're looking at the docs, I think this section could use a rework too. I honestly can't understand the difference in Alignment modes based on this description or the example

felixhabib reviewed (changes requested):
Let's workshop the new example code a little more. Otherwise, looks great!

cherylpaulsen commented:
Hey @felixhabib, I've updated based on our slack chat:
- Updated the 'Providing user controls' section example
- Removed colons from example captions
